ark/utils/notebooks_test.py

```diff
@@ -1,125 +1,567 @@
+    """
+    Tests Notebook 2 - Cluster Pixels for completion.
+    NOTE: When modifying the tests, make sure the test are in the
+    same order as the tagged cells in the notebook.
+    """
+    @pytest.fixture(autouse=True, scope="function")
+    def _setup(self, nb2_context):
+        """
+        Sets up necessary data and paths to run the notebooks.
+        """
+        self.tb: testbook = nb2_context[0]
+        self.base_dir: pathlib.Path = nb2_context[1]
+
+        # Variables
+        self.pixel_prefix = "test"
+        self.channels = ["CD3", "CD4", "CD163_nuc_exclude", "ECAD_smoothed"]
+
+    def test_imports(self):
+        self.tb.execute_cell("import")
+
+    def test_base_dir(self):
+        base_dir_inject = f"""
+            base_dir = "{self.base_dir}"
+        """
+        self.tb.inject(base_dir_inject, "base_dir")
+
+    def test_file_path(self):
+        self.tb.execute_cell("file_path")
+
+    def test_ex_data_download(self):
+        self.tb.execute_cell("ex_data_download")
+
+    def test_load_fovs(self):
+        load_fovs_inject = """
+            fovs = ["fov0", "fov1"]
+        """
+        self.tb.inject(load_fovs_inject, "load_fovs")
+
+    def test_set_multi(self):
+        self.tb.execute_cell("set_multi")
+
+    def test_pixel_prefix(self):
+        pixel_prefix_inject = f"""
+            pixel_cluster_prefix = "{self.pixel_prefix}"
+        """
+        self.tb.inject(pixel_prefix_inject, "pixel_prefix")
+
+    def test_dir_set(self):
+        self.tb.execute_cell("dir_set")
+
+    def test_smooth_channels(self):
+        self.tb.execute_cell("smooth_channels")
+
+    def test_filter_channels(self):
+        self.tb.execute_cell("filter_channels")
+
+    def test_channel_set(self):
+        channel_set_inject = f"""
+            channels = {self.channels}
+            blur_factor = 2
+            subset_proportion = 0.1
+        """
+        self.tb.inject(channel_set_inject, "channel_set")
+
+    def test_gen_pixel_mat(self):
+        self.tb.execute_cell("gen_pixel_mat")
+
+    def test_pixel_som_path_set(self):
+        self.tb.execute_cell("pixel_som_path_set")
+
+    @pytest.mark.skip(reason="Do not train the Pixel SOM")
+    def test_train_pixel_som(self):
+        self.tb.execute_cell("train_pixel_som")
+
+    @pytest.mark.skip(reason="Do not assign Pixel SOM Clusters")
+    def test_cluster_pixel_mat(self):
+        self.tb.execute_cell("cluster_pixel_mat")
+
+    def test_pixel_consensus_cluster(self):
+        # Get Pixel Clustering filenames
+
+        # Get pixel paths and fovs
+        pixel_data_dir = self.tb.ref("pixel_data_dir")
+        pixel_channel_avg_som_cluster = self.tb.ref("pc_chan_avg_som_cluster_name")
+        pixel_channel_avg_meta_cluster = self.tb.ref("pc_chan_avg_meta_cluster_name")
+        fovs = self.tb.ref("fovs")
+        channels = self.tb.ref("channels")
+        # Create fake pixel som files
+        notebooks_test_utils.create_pixel_som_files(self.base_dir,
```

Comment:
Same thing for this and the cell_som function. Is this so that we don't have to actually train the SOM due to save time, or for some other purpose?

Reply:
Yes, this is for skipping the SOM training portion.